app/views/puppetclasses/_form.html.erb

```diff
@@ -1,5 +1,8 @@
 <%= form_for @puppetclass, :url => (@puppetclass.new_record? ? puppetclasses_path : puppetclass_path(:id => @puppetclass)) do |f| %>
   <%= base_errors_for @puppetclass %>
+  <% unless @puppetclass.errors.empty? %>
+    <%= alert :class => 'alert-danger', :header => '', :text => _("The puppet class has not been saved because of an existing error in the class or one of it's parameters.") %>
```

Comment:
nitpick: IMHO, it should be either `"Puppet class has not..."` or `"The class has not..."`.